lib/trento/settings/policy.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Trento.Settings.Policy do
+  @moduledoc """
+  Policy for the Settings resource
+
+  User with the ability all:all can generate a new api key.
+  User with the ability all:api_key_settings can generate a new api key.
+  User with the ability all:all can change activity logs settings.
+  User with the ability all:activity_logs_settings can change activity logs settings.
+  """
+  require Logger
+  @behaviour Bodyguard.Policy
+
+  import Trento.Support.PolicyHelper
+  alias Trento.Settings.ApiKeySettings
+  alias Trento.Users.User
+
+  def authorize(:update_api_key_settings, %User{} = user, ApiKeySettings),
+    do: has_global_ability?(user) or has_api_key_settings_change_ability?(user)
+
+  def authorize(action, %User{}, ApiKeySettings)
+      when action in [:settings, :get_api_key_settings],
+      do: true
+
+  def authorize(:update_activity_log_settings, %User{} = user, ApiKeySettings),
```

Comment:
Please also refactor the matches as the other comments in the software update policy